### ISSUE TITLE
[java] JavaRuleViolation reports wrong class name

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#2212]
**Affects PMD Version:** 6.20.0

**Description:**

In some cases, the reported class is wrong. This might be a regression of #2105 .

**Code Sample demonstrating the issue:**

```java
package pkg;
public class Foo {
    int a;
}
```

If e.g. UnusedField is reported on the node "FieldDeclaration", then the resulting class name is "Foo$Foo" instead of just "Foo".

```xml
<violation beginline="3" endline="3" begincolumn="4" endcolumn="9" rule="UnusedPrivateField" ruleset="Best Practices" package="pkg" class="Foo$Foo" variable="a" externalInfoUrl="https://pmd.github.io/pmd-6.20.0/pmd_rules_java_bestpractices.html#unusedprivatefield" priority="3">
 Avoid unused private fields such as 'a'.
 </violation>
```

Note, that this only affects nodes, the are descendants of the class declaration. If the violation is created directly on the class node, then the class name is fine.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
